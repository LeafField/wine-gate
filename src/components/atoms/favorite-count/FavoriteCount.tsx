import React, { FC } from "react";

type Props = {
  count: number;
};

const FavoriteCount: FC<Props> = ({ count }) => {
  return <p>お気に入り数:{count}</p>;
};

export default FavoriteCount;
