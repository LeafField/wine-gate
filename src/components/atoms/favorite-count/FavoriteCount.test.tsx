import { screen, render } from "@testing-library/react";
import FavoriteCount from "./FavoriteCount";

describe("FavoriteCountの単体テスト", () => {
  test("渡されたPropsに対して正しくレンダリングされているか", () => {
    const { container } = render(<FavoriteCount count={48} />);
    expect(screen.getByText("お気に入り数:48")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
