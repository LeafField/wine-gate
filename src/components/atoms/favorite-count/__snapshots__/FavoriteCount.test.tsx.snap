// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCountの単体テスト 渡されたPropsに対して正しくレンダリングされているか 1`] = `
<div>
  <p>
    お気に入り数:
    48
  </p>
</div>
`;
